classes/files_iterator.php

```diff
@@ -165,11 +165,11 @@ private function extract_context($row) {
     /**
      * @return \stored_file
      */
-    public function current() {
+    public function current(): mixed {
```

Comment:
The problem is that it will break things for anyone on < php8.0

Reply:
Also, type hinting doesn't fix anything for PHP 8.1 - it's still optional in PHP8.1

Reply:
@gthomas2 Since Moodle 4.2 PHP 8.0 is the minimum requirement. I suggest to create a separate branch for MOODLE_402_STABLE (and/or MOODLE_403_STABLE).

Without type hinting in PHP 8.1 you will see failing in Behat tests (e.g. filter/ally/tests/behat/filter.feature) like this:
`When I reload the page: PHP errors found:
Unknown error type: During inheritance of Iterator: Uncaught Exception: Unknown error type: Return type of tool_ally\files_iterator::current() should either be compatible with Iterator::current(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/admin/tool/ally/classes/files_iterator.php on line 168 in /var/www/html/lib/behat/lib.php:157
`

Reply:
Sorry, you are correct here. We actually have a PR open for Moodle 4.2 but we used an annotation to fix the errors: https://github.com/anthology-ally/moodle-tool_ally/pull/91/files#diff-56d533e97b1881bb6228b9b1ca3266c2bb79597f9c57d872608e6e04a46c5701